File: src/java/cat.java
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.channels.FileChannel;

class Cat {
    private static final long BLOCK_SIZE = 131072;

    private static void copyFile( File from, File to ) throws IOException {

        if ( !to.exists() ) { to.createNewFile(); }

        try (
                FileChannel in = new FileInputStream( from ).getChannel();
                FileChannel out = new FileOutputStream( to ).getChannel()
        ) {
            long start = 0;
            long transferred = 0;
            do {
                transferred = out.transferFrom(in, start, BLOCK_SIZE);
                start += transferred;
            } while (transferred > 0);
        }
    }

    public static void main(String[] args) throws IOException {
        final File data = new File("../data");
        final File stdout = new File("/dev/stdout");
        copyFile(data, stdout);
    }
}

File: src/scala/cat.scala
import java.io._

object Cat {
       def read(file: String) {
	   val is = new FileInputStream(file);
	   var n = 0;
	   val buffer = Array.fill[Byte](131072)(0);
	   do {
	   	 n = is.read(buffer);
		 if (n > 0) {
	   	    System.out.write(buffer, 0, n);
		 }
           } while (n > 0);
       }
       def main(args: Array[String]): Unit = {
       	   read("../data");
       }
}
